src/libraries/System.Net.Http/src/System/Net/Http/Headers/WarningHeaderValue.cs

```diff
@@ -328,7 +328,7 @@ private static void CheckAgent(string agent)
 
             // 'receivedBy' can either be a host or a token. Since a token is a valid host, we only verify if the value
             // is a valid host.
-             if (HttpRuleParser.GetHostLength(agent, 0, true, out string? host) != agent.Length)
+             if (HttpRuleParser.GetHostLength(agent, 0, true, out _) != agent.Length)
```

Comment:
Personally, I find all of these unused locals to discards a little frustrating.  IIRC the debugger doesn't show what got written during discards, and from a release build this variable would already be optimized away.  So there's a change here of before you could have this in a debugger in a debug build and F10 might show that the out value was unexpected... now it's a bit more work to obtain that data.

Not a major deal, but it built up enough that I thought I'd say something.